### PR TITLE
[12.0][ADD] helpdesk_type_reminder

### PR DESCRIPTION
**Pull request by ChristianSantamaria**
Add a field includ_reminder in Helpdesk Ticket Type so that when creating a new ticket it places the default value in active_reminder according to the type of ticket.

### CODE REVIEW

File: helpdesk_type_reminder/README.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+========================
+Helpdesk Ticket Reminder
+========================
```
Comment (dalonsod) [inline helpdesk_type_reminder/README.rst]: Please update `README.rst` file

File: helpdesk_type_reminder/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# © 2020 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+{
+    "name": "Helpdesk Ticket Type Reminder",
+    "summary": """
+        Add a field includ_reminder in Helpdesk Ticket Type 
+        so that when creating a new ticket it places the default 
+        value in active_reminder according to the type of ticket.
+    """,
+    "version": "12.0.2.1.0",
```
Comment (dalonsod) [inline helpdesk_type_reminder/__manifest__.py]: Wrong version:
```suggestion
    "version": "12.0.1.0.0",
```

File: helpdesk_type_reminder/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    "license": "LGPL-3",
+    "category": "After-Sales",
+    "author": "Solvos",
+    "website": "https://github.com/solvosci/slv-helpdesk",
+    "depends": [
+        "helpdesk_mgmt",
```
Comment (dalonsod) [inline helpdesk_type_reminder/__manifest__.py]: Wrong dependency:
```suggestion
        "helpdesk_mgmt_ticket_reminder",
```

File: helpdesk_type_reminder/models/helpdesk_ticket_type.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# © 2020 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import fields, models
+
+
+class HelpdeskType(models.Model):
+    _inherit = "helpdesk.ticket.type"
+
+    include_reminder = fields.Boolean(string='Include in reminder')
```
Comment (dalonsod) [inline helpdesk_type_reminder/models/helpdesk_ticket_type.py]: @lmiguens-solvos why should default value be?

**Review by dalonsod — CHANGES_REQUESTED** [review]
Only code review

File: helpdesk_type_reminder/models/helpdesk_ticket_type.py (hunk 4 above)
Comment (lmiguens-solvos) [reply helpdesk_type_reminder/models/helpdesk_ticket_type.py]: I think it should have the same default value as the field reminder_active of helpdesk.ticket (reminder_active = fields.Boolean (string = 'Reminder include', default = True)) right?

File: helpdesk_type_reminder/__manifest__.py [hunk 5]
```diff
@@ -3,17 +3,16 @@
 {
     "name": "Helpdesk Ticket Type Reminder",
     "summary": """
-        Add a field includ_reminder in Helpdesk Ticket Type 
-        so that when creating a new ticket it places the default 
-        value in active_reminder according to the type of ticket.
+        Add a reminder_active field in the helpdesk ticket type
+        so that when creating a new ticket put the default
+        value in reminder_active depending on the type of ticket.
```
Comment (dalonsod) [inline helpdesk_type_reminder/__manifest__.py]: Could you provide a briefer `summary` in manifest file?

File: helpdesk_type_reminder/README.rst [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+Helpdesk Ticket Type Reminder
+========================
```
Comment (dalonsod) [inline helpdesk_type_reminder/README.rst]: I think .rst file will not "compile" until this format is fixed:
```suggestion
=============================
Helpdesk Ticket Type Reminder
=============================
```

File: helpdesk_type_reminder/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    """,
+    "version": "12.0.1.0.0",
+    "license": "LGPL-3",
+    "category": "After-Sales",
+    "author": "Solvos",
+    "website": "https://github.com/solvosci/slv-helpdesk",
+    "depends": [
+        "helpdesk_type",
```
Comment (dalonsod) [inline helpdesk_type_reminder/__manifest__.py:16]: Still incorrecte dependency list

File: helpdesk_type_reminder/readme/DESCRIPTION.rst [hunk 8]
```diff
@@ -0,0 +1,3 @@
+Add a reminder_active field in the helpdesk ticket type
```
Comment (dalonsod) [inline helpdesk_type_reminder/readme/DESCRIPTION.rst]: Typo:
```suggestion
Adds a reminder_active field in the helpdesk ticket type
```

File: helpdesk_type_reminder/views/helpdesk_ticket_views.xml [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+    <record id="ticket_type_view_form" model="ir.ui.view">
+        <field name="name">helpdesk.ticket.type.view.form (reminder_active)</field>
+        <field name="model">helpdesk.ticket.type</field>
+        <field name="inherit_id" ref="helpdesk_type.helpdesk_ticket_type_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='team_ids']" position="after">
+                <field name="reminder_active"/>
+            </xpath>
+        </field>
+    </record>
```
Comment (dalonsod) [inline helpdesk_type_reminder/views/helpdesk_ticket_views.xml:12]: Could you add a tree and search view with references to the new field?

File: helpdesk_type_reminder/__manifest__.py [hunk 10]
```diff
@@ -0,0 +1,21 @@
+# © 2020 Solvos Consultoría Informática (<http://www.solvos.es>)
```
Comment (dalonsod) [inline helpdesk_type_reminder/__manifest__.py]: Wrong year:
```suggestion
# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
```

File: helpdesk_type_reminder/models/helpdesk_ticket.py [hunk 11]
```diff
@@ -0,0 +1,12 @@
+# © 2020 Solvos Consultoría Informática (<http://www.solvos.es>)
```
Comment (dalonsod) [inline helpdesk_type_reminder/models/helpdesk_ticket.py]: Wrong year:
```suggestion
# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
```

File: helpdesk_type_reminder/models/helpdesk_ticket_type.py [hunk 12]
```diff
@@ -0,0 +1,9 @@
+# © 2020 Solvos Consultoría Informática (<http://www.solvos.es>)
```
Comment (dalonsod) [inline helpdesk_type_reminder/models/helpdesk_ticket_type.py]: Wrong year:
```suggestion
# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
```

**Review by dalonsod — CHANGES_REQUESTED** [review]
See new comments

File: helpdesk_type_reminder/__manifest__.py [hunk 13]
```diff
@@ -0,0 +1,22 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+{
+    "name": "Helpdesk Ticket Type Reminder",
+    "summary": """
+        Adds a reminder_active field in the helpdesk ticket type
+        sets default reminder active flag for a ticket depending 
+        on the selected type
```
Comment (dalonsod) [inline helpdesk_type_reminder/__manifest__.py]: Please don't forget changing to a simpler `summary`

File: helpdesk_type_reminder/views/helpdesk_ticket_views.xml (hunk 9 above)
Comment (dalonsod) [reply helpdesk_type_reminder/views/helpdesk_ticket_views.xml:12]: Still missing added field to tree view

File: helpdesk_type_reminder/views/helpdesk_ticket_views.xml [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+                        name="reminder_active_false" 
+                        domain="[('reminder_active','=', False)]"/>
+            </search>
+        </field>
+    </record>
+
+
+</odoo>
+
+
```
Comment (dalonsod) [inline helpdesk_type_reminder/views/helpdesk_ticket_views.xml]: Proper EOF:
```suggestion
</odoo>

```

File: helpdesk_type_reminder/views/helpdesk_ticket_views.xml [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+                <filter string="With Reminder include" 
+                        name="reminder_active_true" 
+                        domain="[('reminder_active','=', True)]"/>
+                <filter string="Without Reminder include" 
+                        name="reminder_active_false" 
+                        domain="[('reminder_active','=', False)]"/>
+            </search>
+        </field>
+    </record>
+
```
Comment (dalonsod) [inline helpdesk_type_reminder/views/helpdesk_ticket_views.xml]: Remove duplicated empty line

**Review by dalonsod — CHANGES_REQUESTED** [review]
Locally tested, it works!
Please attend comments, squash and I'll merge it

**Review by dalonsod — APPROVED** [review]
LGTM, good work
@lmiguens-solvos do you agree?

**Review by lmiguens-solvos — CHANGES_REQUESTED** [review]
Good work. Tested in my environment and works correctly.
But I detected that some translations are missing, that of the added filters in ticket_type_search_view

**Review by lmiguens-solvos — APPROVED** [review]

**Comment by dalonsod** [comment]
OK, merging...

**Pull request merged**